src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,767 @@
+                            }
+                            if (tilePos.y == bottomLeft.y)
+                            {
+                                wallArgs.rotation = 3;
+                                bool placeEntrance = buildingWallEntranceMask & (1ULL << 0);
+                                buildingWallEntranceMask = Numerics::ror(buildingWallEntranceMask, 1);
+                                wallArgs.type = placeEntrance ? indObj->buildingWallEntrance : indObj->buildingWall;
+                                doCommand(wallArgs, Flags::apply);
+                            }
+                            if (tilePos.y == topRight.y)
+                            {
+                                wallArgs.rotation = 1;
+                                bool placeEntrance = buildingWallEntranceMask & (1ULL << 0);
+                                buildingWallEntranceMask = Numerics::ror(buildingWallEntranceMask, 1);
+                                wallArgs.type = placeEntrance ? indObj->buildingWallEntrance : indObj->buildingWall;
+                                doCommand(wallArgs, Flags::apply);
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Expand grounds
+            if (indObj->var_EC != 0)
+            {
+                const auto numExpands = (((indObj->var_EB * newIndustry->prng.randNext(0xFF)) / 256) + 1) * 4;
+                for (auto i = 0; i < numExpands; ++i)
+                {
+                    const auto randExpandVal = newIndustry->prng.randNext();
+                    // dl
+                    const auto surfaceUnk = (((randExpandVal & 0xFF) * indObj->var_EC) / 256) | (((randExpandVal >> 8) & 0x7) << 5);
+
+                    const World::TilePos2 randOffset(
+                        ((randExpandVal >> 11) & 0x1F) - 15,
+                        ((randExpandVal >> 16) & 0x1F) - 15);
+
+                    const World::Pos2 randPos = World::Pos2{ newIndustry->x, newIndustry->y } + World::toWorldSpace(randOffset);
+
+                    bool useSecondWallType = ((randExpandVal >> 21) & 1) && indObj->wallTypes[2] != 0xFF;
+
+                    const auto wallType = useSecondWallType ? indObj->wallTypes[2] : indObj->wallTypes[0];
+                    const auto wallEntranceType = useSecondWallType ? indObj->wallTypes[3] : indObj->wallTypes[1];
+                    newIndustry->expandGrounds(randPos, wallType, wallEntranceType, surfaceUnk);
+                }
+            }
+        }
+
+        // Cleanup
+        if (!(flags & Flags::apply))
+        {
+            StringManager::emptyUserString(newIndustry->name);
+            // Free the industry slot
+            newIndustry->name = StringIds::null;
+        }
+        totalCost += Economy::getInflationAdjustedCost(indObj->costFactor, indObj->costIndex, 3);
+
+        // Send message post
+        if ((flags & Flags::apply) && !(flags & Flags::flag_6) && !args.buildImmediately)
```

Comment:
Starting to believe that Flags::flag_6 is for ghosts, RCT2 also has it there. But that can be renamed afterwards, just a remark.

Reply:
Yeah i keep putting off renaming it but it's getting ridiculous now